backend/workers/expire_items.py

```diff
@@ -62,6 +62,7 @@ def expire_datasets(self):
 				dataset = DataSet(key=dataset["key"], db=self.db)
 				if dataset.is_expired():
 					self.log.info(f"Deleting dataset {dataset.key} (expired)")
+					dataset.delete_annotations(dataset_key=dataset.key)
```

Comment:
Wouldn't it make sense to add `delete_annotations()` to `delete()` instead? You probably always want to delete the annotations too if you're deleting a dataset